### ISSUE TITLE
The inputType "String!" of a field in an annotated input type is being ignored when update is set to true.

### ISSUE THREAD

**Issue by hjerichen** [thecodingmachine/graphqlite#371]
When an input type ist specified with `update: true`, it makes all fields optional (as stated in the documentation).
It even does it when I specifically say it should be non optional through the inputType in a Field Annotation.

```php
#[Input(update: true)]
class UserInput
{
    #[Field(inputType: 'String!')]
    public string $important;
}
```
The output will be `String` and not `String!`.

Is this intended behavior?
I would expect that my specified input type is used.

**Comment by Lappihuan** [comment]
As explained by @devmaslov [here](https://github.com/thecodingmachine/graphqlite/pull/269#issue-634866663), this is the intended behavior.
I will check if its easy to implement.

**Comment by oojacoboo** [comment]
@Lappihuan The example given here should default to `String!` anyway, since it's not nullable typed in PHP, regardless of the annotation.

**Comment by Lappihuan** [comment]
@oojacoboo no, update makes everything nullable regardless of what php says.
this is actually intended and a really cool feature as described [here](https://graphqlite.thecodingmachine.io/docs/input-types#multiple-input-types-per-one-class)
```
Note that update: true argument for UpdateUserInput. It should be used when input type is used for a partial update, It makes all fields optional and removes all default values from thus prevents setting default values via setters or directly to public properties. In example above if you use the class as UpdateUserInput and set only username the other ones will be ignored. In PHP 7 they will be set to null, while in PHP 8 they will be in not initialized state - this can be used as a trick to check if user actually passed a value for a certain field.
```

**Comment by oojacoboo** [comment]
@Lappihuan yep, overlooked that it was an `update` here.  I guess this is the right approach here.  I do wonder if we'd benefit from another attribute on the `Field` annotation to make it "required".  Or maybe that's at least worth looking into/considering.

**Comment by Lappihuan** [comment]
@oojacoboo i think type forcing as tested above should be fine, it would follow the current workflow of how to overwrite something between PHP types and GQL Types.

**Comment by Lappihuan** [comment]
This would be supported with this commit:
https://github.com/thecodingmachine/graphqlite/pull/466/commits/1a21b830880cb68429e33e1c841392b624627bc6

**Comment by oojacoboo** [comment]
Closing - assuming this may be resolved in version 6.0.  If not, please reopen.
